src/main/java/com/dcaiti/mosaic/app/tse/persistence/FcdDatabaseHelper.java

```diff
@@ -267,19 +266,21 @@ public void insertFcdRecords(String vehicleId, Collection<FcdRecord> records) {
         try (PreparedStatement statement = connection.prepareStatement(sqlRecordInsert)) {
             connection.setAutoCommit(false);
             int i = 0;
-            for (FcdRecord record : records) {
-                statement.setString(1, vehicleId);
-                statement.setLong(2, record.getTimeStamp());
-                statement.setDouble(3, record.getPosition().getLatitude());
-                statement.setDouble(4, record.getPosition().getLongitude());
-                statement.setString(5, record.getConnectionId());
-                statement.setDouble(6, record.getOffset());
-                statement.setDouble(7, record.getSpeed());
-                statement.addBatch();
-                i++;
-                if (i % 1000 == 0 || i == records.size()) {
-                    statement.executeBatch();
-                    connection.commit();
+            for (String vehicleId : records.keySet()) {
+                for (FcdRecord record : records.get(vehicleId)) {
```

Comment:
```
for (Entry recordEntry: records.entrySet()) {
  for (FcdRector record: recordsEntry.getValue()) {
    ...
  }
}
```

is more efficient

Reply:
I pushed the respective changes and this PR can be merged from my side.